src/ripple/ledger/impl/PaymentSandbox.cpp

```diff
@@ -172,7 +172,11 @@ PaymentSandbox::balanceHook (AccountID const& account,
         }
     }
 
-    assert (!isXRP(issuer) || adjustedAmt >= beast::zero);
+    if (isXRP(issuer) && adjustedAmt < beast::zero)
+    {
+        adjustedAmt.clear();
```

Comment:
I take it that a negative adjusted XRP amount is nothing to worry about?  It would be great to drop in a comment saying that along with a justification.